### ISSUE TITLE
bookmarking selectInput

### ISSUE THREAD

**Issue by KZARCA** [rstudio/shiny#2305]
Considering this simple app:
```r
ui <- function(request) {
  fluidPage(
           uiOutput("test"),
           bookmarkButton()
        )
}

server <- function(session, input, output) {
   output$test <- renderUI({
     selectInput("foo", label = NULL, choices = letters[1:10], 
                 selected = if (!is.null(input$foo)) input$foo)
   })
}

shinyApp(ui = ui, server = server, enableBookmarking = "url", port = 4935)
```
If I select any value for "foo", bookmark and then restore, I cannot change the value (it remains the same as when it was bookmarked).

**Comment by alandipert** [comment]
@KZARCA Thanks for the report.

@jcheng5 This one bisects to 0bb53e8ca5b8e769b11a0ec37a5b5c66eceb9a54 also

**Comment by wch** [comment]
Fixed in the rc-v1.3.0 branch by #2360.
